src/network/websockets/server/index.ts

```diff
@@ -112,6 +112,13 @@ export function runWebsocketServer(params: Params) {
     };
 
     socket.on("message", function(incomingBytes) {
+      try {
+        JSON.parse(incomingBytes.toString());
```

Comment:
I am wary of repeatin the parsing logic twice. It becomes a risk of bugs if we adjust one, but not the other.

```ts
// Before
try {
  JSON.parse(incomingBytes.toString());
 } catch (err) {
  console.warn("Received invalid message format");
  socket.terminate();
  return;
}
const msg = JSON.parse(incomingBytes.toString());

// After
let msg: string;
try {
  msg = JSON.parse(incomingBytes.toString());
 } catch (err) {
  console.warn("Received invalid message format");
  socket.terminate();
  return;
}

// Arguably better
function attempt<T>(f: () => T): T | undefined {
  try { return f(); } catch (err) { return undefined; }
}
const msg = attempt(() => JSON.parse(incomingBytes.toString()));
if (!msg) {
  console.warn("Received invalid message format");
  socket.terminate();
  return;
}
```